cmd/deploy/deploy.go

```diff
@@ -112,25 +124,69 @@ func main() {
 	headerReaderConfig := headerreader.DefaultConfig
 	headerReaderConfig.TxTimeout = *txTimeout
 
-	deployPtr, err := arbnode.DeployOnL1(
+	chainConfigJson, err := os.ReadFile(*l2ChainConfig)
+	if err != nil {
+		panic(fmt.Errorf("failed to read l2 chain config file: %w", err))
+	}
+	var chainConfig params.ChainConfig
+	err = json.Unmarshal(chainConfigJson, &chainConfig)
+	if err != nil {
+		panic(fmt.Errorf("failed to deserialize chain config: %w", err))
+	}
+
+	if chainConfig.ChainID.Cmp(l2ChainId) != 0 {
+		panic(fmt.Sprintf("chain id mismatch, id from args: %v, id from l2 chain config: %v", l2ChainId, chainConfig.ChainID))
```

Comment:
Let's just load the l2 chain id from this file instead of specifying it on the CLI.